{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook is to create the preparation information and sample information files for Qiita, as well as the mapping file for Qiime."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## create preparation_info document"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "info = pd.read_csv('../data/Demultiplex_Sheet.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#SampleID</th>\n",
       "      <th>BarcodeSequence</th>\n",
       "      <th>LinkerPrimerSequence</th>\n",
       "      <th>ReversePrimer</th>\n",
       "      <th>Description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>Orwoll.BI0023.BI</td>\n",
       "      <td>TCTGGTGACATT</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>GTGCCAGCMGCCGCGGTAA</td>\n",
       "      <td>Orwoll.BI0023.BI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1129</th>\n",
       "      <td>Orwoll.BI0023.BI</td>\n",
       "      <td>TCTGGTGACATT</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>GTGCCAGCMGCCGCGGTAA</td>\n",
       "      <td>Orwoll.BI0023.BI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>649</th>\n",
       "      <td>Orwoll.BI0056.BI</td>\n",
       "      <td>CAAGCATGCCTA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>GTGCCAGCMGCCGCGGTAA</td>\n",
       "      <td>Orwoll.BI0056.BI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>475</th>\n",
       "      <td>Orwoll.BI0056.BI</td>\n",
       "      <td>CAAGCATGCCTA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>GTGCCAGCMGCCGCGGTAA</td>\n",
       "      <td>Orwoll.BI0056.BI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>Orwoll.BI0131.BI</td>\n",
       "      <td>CTATTTGCGACA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>GTGCCAGCMGCCGCGGTAA</td>\n",
       "      <td>Orwoll.BI0131.BI</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             #SampleID BarcodeSequence  LinkerPrimerSequence  \\\n",
       "60    Orwoll.BI0023.BI    TCTGGTGACATT  GGACTACHVGGGTWTCTAAT   \n",
       "1129  Orwoll.BI0023.BI    TCTGGTGACATT  GGACTACHVGGGTWTCTAAT   \n",
       "649   Orwoll.BI0056.BI    CAAGCATGCCTA  GGACTACHVGGGTWTCTAAT   \n",
       "475   Orwoll.BI0056.BI    CAAGCATGCCTA  GGACTACHVGGGTWTCTAAT   \n",
       "75    Orwoll.BI0131.BI    CTATTTGCGACA  GGACTACHVGGGTWTCTAAT   \n",
       "\n",
       "            ReversePrimer       Description  \n",
       "60    GTGCCAGCMGCCGCGGTAA  Orwoll.BI0023.BI  \n",
       "1129  GTGCCAGCMGCCGCGGTAA  Orwoll.BI0023.BI  \n",
       "649   GTGCCAGCMGCCGCGGTAA  Orwoll.BI0056.BI  \n",
       "475   GTGCCAGCMGCCGCGGTAA  Orwoll.BI0056.BI  \n",
       "75    GTGCCAGCMGCCGCGGTAA  Orwoll.BI0131.BI  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "info = info.sort_values(['#SampleID'])\n",
    "info.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "info = info.drop_duplicates(subset=['#SampleID', 'ReversePrimer'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#SampleID</th>\n",
       "      <th>BarcodeSequence</th>\n",
       "      <th>LinkerPrimerSequence</th>\n",
       "      <th>ReversePrimer</th>\n",
       "      <th>Description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>Orwoll.BI0023.BI</td>\n",
       "      <td>TCTGGTGACATT</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>GTGCCAGCMGCCGCGGTAA</td>\n",
       "      <td>Orwoll.BI0023.BI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>649</th>\n",
       "      <td>Orwoll.BI0056.BI</td>\n",
       "      <td>CAAGCATGCCTA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>GTGCCAGCMGCCGCGGTAA</td>\n",
       "      <td>Orwoll.BI0056.BI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>Orwoll.BI0131.BI</td>\n",
       "      <td>CTATTTGCGACA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>GTGCCAGCMGCCGCGGTAA</td>\n",
       "      <td>Orwoll.BI0131.BI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>932</th>\n",
       "      <td>Orwoll.BI0153.BI</td>\n",
       "      <td>ATCGGCGTTACA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>GTGCCAGCMGCCGCGGTAA</td>\n",
       "      <td>Orwoll.BI0153.BI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>833</th>\n",
       "      <td>Orwoll.BI0215.BI</td>\n",
       "      <td>CCTCTCGTGATC</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>GTGCCAGCMGCCGCGGTAA</td>\n",
       "      <td>Orwoll.BI0215.BI</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            #SampleID BarcodeSequence  LinkerPrimerSequence  \\\n",
       "60   Orwoll.BI0023.BI    TCTGGTGACATT  GGACTACHVGGGTWTCTAAT   \n",
       "649  Orwoll.BI0056.BI    CAAGCATGCCTA  GGACTACHVGGGTWTCTAAT   \n",
       "75   Orwoll.BI0131.BI    CTATTTGCGACA  GGACTACHVGGGTWTCTAAT   \n",
       "932  Orwoll.BI0153.BI    ATCGGCGTTACA  GGACTACHVGGGTWTCTAAT   \n",
       "833  Orwoll.BI0215.BI    CCTCTCGTGATC  GGACTACHVGGGTWTCTAAT   \n",
       "\n",
       "           ReversePrimer       Description  \n",
       "60   GTGCCAGCMGCCGCGGTAA  Orwoll.BI0023.BI  \n",
       "649  GTGCCAGCMGCCGCGGTAA  Orwoll.BI0056.BI  \n",
       "75   GTGCCAGCMGCCGCGGTAA  Orwoll.BI0131.BI  \n",
       "932  GTGCCAGCMGCCGCGGTAA  Orwoll.BI0153.BI  \n",
       "833  GTGCCAGCMGCCGCGGTAA  Orwoll.BI0215.BI  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "info.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "600"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "info.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "n = info.shape[0]\n",
    "data = {'Sample_Name': info.loc[:,'#SampleID'].str[7:13].values, # .values takes only value not index\n",
    "       'Barcode': info.loc[:,'BarcodeSequence'].values,\n",
    "       'LinkerPrimerSequence': info.loc[:, 'LinkerPrimerSequence'].values,\n",
    "       'Description': info.loc[:,'Description'].values,\n",
    "       'Experiment_Design_Description': np.repeat(\n",
    "           '16S stool samples sequenced for MrOS Vitamin D study', n),\n",
    "       'Library_Construction_Protocol': np.repeat('16S rRNA v4', n),\n",
    "       'Linker': np.repeat('GT', n),\n",
    "       'Platform': np.repeat('Illumina', n),\n",
    "       'Center_Name': info.loc[:,'#SampleID'].str[14:16].values,\n",
    "       'Center_Project': np.repeat('MrOS', n),\n",
    "       'Instrument_Model': np.repeat('Illumina MiSeq', n)}\n",
    "prep_info = pd.DataFrame(data, columns=['Sample_Name', 'Barcode', 'LinkerPrimerSequence', \n",
    "                                        'Description', \n",
    "                                        'Experiment_Design_Description',\n",
    "                                        'Library_Construction_Protocol', 'Linker', 'Platform',\n",
    "                                        'Center_Name', 'Center_Project', 'Instrument_Model'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(600, 11)"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prep_info.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Name</th>\n",
       "      <th>Barcode</th>\n",
       "      <th>LinkerPrimerSequence</th>\n",
       "      <th>Description</th>\n",
       "      <th>Experiment_Design_Description</th>\n",
       "      <th>Library_Construction_Protocol</th>\n",
       "      <th>Linker</th>\n",
       "      <th>Platform</th>\n",
       "      <th>Center_Name</th>\n",
       "      <th>Center_Project</th>\n",
       "      <th>Instrument_Model</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BI0023</td>\n",
       "      <td>TCTGGTGACATT</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0023.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BI0056</td>\n",
       "      <td>CAAGCATGCCTA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0056.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BI0131</td>\n",
       "      <td>CTATTTGCGACA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0131.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BI0153</td>\n",
       "      <td>ATCGGCGTTACA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0153.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BI0215</td>\n",
       "      <td>CCTCTCGTGATC</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0215.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>BI0353</td>\n",
       "      <td>TGCCATCTGAAT</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0353.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>BI0371</td>\n",
       "      <td>GAATAGAGCCAA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0371.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>BI0372</td>\n",
       "      <td>ATGCCAACCAAC</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0372.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>BI0380</td>\n",
       "      <td>GAGTCGCGTTTA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0380.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>BI0389</td>\n",
       "      <td>GCTATGGACCGA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0389.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Sample_Name       Barcode  LinkerPrimerSequence       Description  \\\n",
       "0      BI0023  TCTGGTGACATT  GGACTACHVGGGTWTCTAAT  Orwoll.BI0023.BI   \n",
       "1      BI0056  CAAGCATGCCTA  GGACTACHVGGGTWTCTAAT  Orwoll.BI0056.BI   \n",
       "2      BI0131  CTATTTGCGACA  GGACTACHVGGGTWTCTAAT  Orwoll.BI0131.BI   \n",
       "3      BI0153  ATCGGCGTTACA  GGACTACHVGGGTWTCTAAT  Orwoll.BI0153.BI   \n",
       "4      BI0215  CCTCTCGTGATC  GGACTACHVGGGTWTCTAAT  Orwoll.BI0215.BI   \n",
       "5      BI0353  TGCCATCTGAAT  GGACTACHVGGGTWTCTAAT  Orwoll.BI0353.BI   \n",
       "6      BI0371  GAATAGAGCCAA  GGACTACHVGGGTWTCTAAT  Orwoll.BI0371.BI   \n",
       "7      BI0372  ATGCCAACCAAC  GGACTACHVGGGTWTCTAAT  Orwoll.BI0372.BI   \n",
       "8      BI0380  GAGTCGCGTTTA  GGACTACHVGGGTWTCTAAT  Orwoll.BI0380.BI   \n",
       "9      BI0389  GCTATGGACCGA  GGACTACHVGGGTWTCTAAT  Orwoll.BI0389.BI   \n",
       "\n",
       "                       Experiment_Design_Description  \\\n",
       "0  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "1  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "2  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "3  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "4  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "5  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "6  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "7  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "8  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "9  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "\n",
       "  Library_Construction_Protocol Linker  Platform Center_Name Center_Project  \\\n",
       "0                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "1                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "2                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "3                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "4                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "5                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "6                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "7                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "8                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "9                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "\n",
       "  Instrument_Model  \n",
       "0   Illumina MiSeq  \n",
       "1   Illumina MiSeq  \n",
       "2   Illumina MiSeq  \n",
       "3   Illumina MiSeq  \n",
       "4   Illumina MiSeq  \n",
       "5   Illumina MiSeq  \n",
       "6   Illumina MiSeq  \n",
       "7   Illumina MiSeq  \n",
       "8   Illumina MiSeq  \n",
       "9   Illumina MiSeq  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prep_info.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# fix mismatch sample name 'BI0778' should be 'BIO778' (Nora's email on 08/07/2017)\n",
    "prep_info = prep_info.replace(to_replace='BIO778', value='BI0778')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create sample_info document"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "samples_part1 = pd.read_csv('../data/VitDMetadata_update.csv', sep=',')\n",
    "samples_part2 = pd.read_csv('../data/Other32metadata.csv', sep=',')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(567, 26)\n",
      "(32, 26)\n"
     ]
    }
   ],
   "source": [
    "print(samples_part1.shape)\n",
    "print(samples_part2.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['ID', 'GIERACE', 'SITE', 'TUDRAMT', 'HWWGT', 'V4AGE1', 'HWHGT',\n",
      "       'PASCORE', 'HWBMI', 'TURSMOKE', 'DTVITD', 'M1ADEPR', 'M1VITMND',\n",
      "       'M1ANTIB', 'M1PROBI', 'OHV1D3', 'OHV24D3', 'OHVD3', 'OHVD2', 'OHV1D2',\n",
      "       'OHV1D2CT', 'OHVD2CT', 'OHVDTOT', 'OHV1DTOT', 'OHSEAS', 'VDstatus'],\n",
      "      dtype='object')\n",
      "Index(['ID', 'GIERACE', 'SITE', 'TUDRAMT', 'HWWGT', 'V4AGE1', 'HWHGT',\n",
      "       'PASCORE', 'HWBMI', 'TURSMOKE', 'DTVITD', 'M1ADEPR', 'M1VITMND',\n",
      "       'M1ANTIB', 'M1PROBI', 'OHV1D3', 'OHV24D3', 'OHVD3', 'OHVD2', 'OHV1D2',\n",
      "       'OHV1D2CT', 'OHVD2CT', 'OHVDTOT', 'OHV1DTOT', 'OHSEAS', 'VDstatus'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "print(samples_part1.columns)\n",
    "print(samples_part2.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# merge two metadata files\n",
    "samples = pd.concat([samples_part1, samples_part2], ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(599, 26)"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "samples.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "599"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(pd.unique(samples.ID))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>GIERACE</th>\n",
       "      <th>SITE</th>\n",
       "      <th>TUDRAMT</th>\n",
       "      <th>HWWGT</th>\n",
       "      <th>V4AGE1</th>\n",
       "      <th>HWHGT</th>\n",
       "      <th>PASCORE</th>\n",
       "      <th>HWBMI</th>\n",
       "      <th>TURSMOKE</th>\n",
       "      <th>...</th>\n",
       "      <th>OHV24D3</th>\n",
       "      <th>OHVD3</th>\n",
       "      <th>OHVD2</th>\n",
       "      <th>OHV1D2</th>\n",
       "      <th>OHV1D2CT</th>\n",
       "      <th>OHVD2CT</th>\n",
       "      <th>OHVDTOT</th>\n",
       "      <th>OHV1DTOT</th>\n",
       "      <th>OHSEAS</th>\n",
       "      <th>VDstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BI0023</td>\n",
       "      <td>1:WHITE</td>\n",
       "      <td>Birmingham</td>\n",
       "      <td>1: Less than one drink per week</td>\n",
       "      <td>83.1</td>\n",
       "      <td>83</td>\n",
       "      <td>169.60</td>\n",
       "      <td>91.000000</td>\n",
       "      <td>28.890119</td>\n",
       "      <td>M:Not Applicable</td>\n",
       "      <td>...</td>\n",
       "      <td>1.77</td>\n",
       "      <td>25.8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>25.8</td>\n",
       "      <td>39.3</td>\n",
       "      <td>3:SUMMER</td>\n",
       "      <td>sufficiency</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BI0056</td>\n",
       "      <td>1:WHITE</td>\n",
       "      <td>Birmingham</td>\n",
       "      <td>0:None drinker</td>\n",
       "      <td>76.2</td>\n",
       "      <td>81</td>\n",
       "      <td>163.40</td>\n",
       "      <td>199.178571</td>\n",
       "      <td>28.539796</td>\n",
       "      <td>1:PAST</td>\n",
       "      <td>...</td>\n",
       "      <td>3.91</td>\n",
       "      <td>39.2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>39.2</td>\n",
       "      <td>61.9</td>\n",
       "      <td>2:SPRING</td>\n",
       "      <td>sufficiency</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BI0131</td>\n",
       "      <td>1:WHITE</td>\n",
       "      <td>Birmingham</td>\n",
       "      <td>0:None drinker</td>\n",
       "      <td>78.5</td>\n",
       "      <td>83</td>\n",
       "      <td>177.15</td>\n",
       "      <td>161.714286</td>\n",
       "      <td>25.014242</td>\n",
       "      <td>1:PAST</td>\n",
       "      <td>...</td>\n",
       "      <td>1.49</td>\n",
       "      <td>23.1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>23.1</td>\n",
       "      <td>52.1</td>\n",
       "      <td>2:SPRING</td>\n",
       "      <td>sufficiency</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BI0153</td>\n",
       "      <td>1:WHITE</td>\n",
       "      <td>Birmingham</td>\n",
       "      <td>4: 6-13 drinks per week</td>\n",
       "      <td>95.1</td>\n",
       "      <td>79</td>\n",
       "      <td>175.50</td>\n",
       "      <td>88.214286</td>\n",
       "      <td>30.876373</td>\n",
       "      <td>1:PAST</td>\n",
       "      <td>...</td>\n",
       "      <td>2.14</td>\n",
       "      <td>27.3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>27.3</td>\n",
       "      <td>43.1</td>\n",
       "      <td>2:SPRING</td>\n",
       "      <td>sufficiency</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BI0215</td>\n",
       "      <td>1:WHITE</td>\n",
       "      <td>Birmingham</td>\n",
       "      <td>3: 3-5 drinks per week</td>\n",
       "      <td>93.0</td>\n",
       "      <td>81</td>\n",
       "      <td>166.40</td>\n",
       "      <td>256.821429</td>\n",
       "      <td>33.587394</td>\n",
       "      <td>1:PAST</td>\n",
       "      <td>...</td>\n",
       "      <td>3.62</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>33</td>\n",
       "      <td>50.2</td>\n",
       "      <td>4:FALL</td>\n",
       "      <td>sufficiency</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>BI0353</td>\n",
       "      <td>1:WHITE</td>\n",
       "      <td>Birmingham</td>\n",
       "      <td>0:None drinker</td>\n",
       "      <td>83.6</td>\n",
       "      <td>80</td>\n",
       "      <td>177.90</td>\n",
       "      <td>179.571429</td>\n",
       "      <td>26.415229</td>\n",
       "      <td>1:PAST</td>\n",
       "      <td>...</td>\n",
       "      <td>1.79</td>\n",
       "      <td>19.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>19.5</td>\n",
       "      <td>45.5</td>\n",
       "      <td>2:SPRING</td>\n",
       "      <td>deficiency</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>BI0371</td>\n",
       "      <td>2:AFRICAN AMERICAN</td>\n",
       "      <td>Birmingham</td>\n",
       "      <td>3: 3-5 drinks per week</td>\n",
       "      <td>102.6</td>\n",
       "      <td>79</td>\n",
       "      <td>190.10</td>\n",
       "      <td>144.571429</td>\n",
       "      <td>28.391159</td>\n",
       "      <td>0:NO</td>\n",
       "      <td>...</td>\n",
       "      <td>1.82</td>\n",
       "      <td>22.3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>22.3</td>\n",
       "      <td>60</td>\n",
       "      <td>2:SPRING</td>\n",
       "      <td>sufficiency</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>BI0372</td>\n",
       "      <td>1:WHITE</td>\n",
       "      <td>Birmingham</td>\n",
       "      <td>1: Less than one drink per week</td>\n",
       "      <td>85.3</td>\n",
       "      <td>80</td>\n",
       "      <td>175.90</td>\n",
       "      <td>91.678571</td>\n",
       "      <td>27.568768</td>\n",
       "      <td>0:NO</td>\n",
       "      <td>...</td>\n",
       "      <td>3.37</td>\n",
       "      <td>37.2</td>\n",
       "      <td>0.7</td>\n",
       "      <td>0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>0: No</td>\n",
       "      <td>37.9</td>\n",
       "      <td>53.6</td>\n",
       "      <td>2:SPRING</td>\n",
       "      <td>sufficiency</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>BI0380</td>\n",
       "      <td>1:WHITE</td>\n",
       "      <td>Birmingham</td>\n",
       "      <td>0:None drinker</td>\n",
       "      <td>110.6</td>\n",
       "      <td>85</td>\n",
       "      <td>192.35</td>\n",
       "      <td>65.607143</td>\n",
       "      <td>29.893086</td>\n",
       "      <td>1:PAST</td>\n",
       "      <td>...</td>\n",
       "      <td>2.55</td>\n",
       "      <td>45.7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>45.7</td>\n",
       "      <td>50.2</td>\n",
       "      <td>2:SPRING</td>\n",
       "      <td>sufficiency</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>BI0389</td>\n",
       "      <td>1:WHITE</td>\n",
       "      <td>Birmingham</td>\n",
       "      <td>0:None drinker</td>\n",
       "      <td>72.2</td>\n",
       "      <td>85</td>\n",
       "      <td>177.70</td>\n",
       "      <td>209.714286</td>\n",
       "      <td>22.864533</td>\n",
       "      <td>0:NO</td>\n",
       "      <td>...</td>\n",
       "      <td>1.93</td>\n",
       "      <td>26.7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>26.7</td>\n",
       "      <td>49.7</td>\n",
       "      <td>2:SPRING</td>\n",
       "      <td>sufficiency</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       ID             GIERACE        SITE                          TUDRAMT  \\\n",
       "0  BI0023             1:WHITE  Birmingham  1: Less than one drink per week   \n",
       "1  BI0056             1:WHITE  Birmingham                   0:None drinker   \n",
       "2  BI0131             1:WHITE  Birmingham                   0:None drinker   \n",
       "3  BI0153             1:WHITE  Birmingham          4: 6-13 drinks per week   \n",
       "4  BI0215             1:WHITE  Birmingham           3: 3-5 drinks per week   \n",
       "5  BI0353             1:WHITE  Birmingham                   0:None drinker   \n",
       "6  BI0371  2:AFRICAN AMERICAN  Birmingham           3: 3-5 drinks per week   \n",
       "7  BI0372             1:WHITE  Birmingham  1: Less than one drink per week   \n",
       "8  BI0380             1:WHITE  Birmingham                   0:None drinker   \n",
       "9  BI0389             1:WHITE  Birmingham                   0:None drinker   \n",
       "\n",
       "   HWWGT  V4AGE1   HWHGT     PASCORE      HWBMI          TURSMOKE  ...  \\\n",
       "0   83.1      83  169.60   91.000000  28.890119  M:Not Applicable  ...   \n",
       "1   76.2      81  163.40  199.178571  28.539796            1:PAST  ...   \n",
       "2   78.5      83  177.15  161.714286  25.014242            1:PAST  ...   \n",
       "3   95.1      79  175.50   88.214286  30.876373            1:PAST  ...   \n",
       "4   93.0      81  166.40  256.821429  33.587394            1:PAST  ...   \n",
       "5   83.6      80  177.90  179.571429  26.415229            1:PAST  ...   \n",
       "6  102.6      79  190.10  144.571429  28.391159              0:NO  ...   \n",
       "7   85.3      80  175.90   91.678571  27.568768              0:NO  ...   \n",
       "8  110.6      85  192.35   65.607143  29.893086            1:PAST  ...   \n",
       "9   72.2      85  177.70  209.714286  22.864533              0:NO  ...   \n",
       "\n",
       "   OHV24D3 OHVD3 OHVD2 OHV1D2 OHV1D2CT OHVD2CT OHVDTOT OHV1DTOT    OHSEAS  \\\n",
       "0     1.77  25.8     0      0   1: Yes  1: Yes    25.8     39.3  3:SUMMER   \n",
       "1     3.91  39.2     0      0   1: Yes  1: Yes    39.2     61.9  2:SPRING   \n",
       "2     1.49  23.1     0      0   1: Yes  1: Yes    23.1     52.1  2:SPRING   \n",
       "3     2.14  27.3     0      0   1: Yes  1: Yes    27.3     43.1  2:SPRING   \n",
       "4     3.62    33     0      0   1: Yes  1: Yes      33     50.2    4:FALL   \n",
       "5     1.79  19.5     0      0   1: Yes  1: Yes    19.5     45.5  2:SPRING   \n",
       "6     1.82  22.3     0      0   1: Yes  1: Yes    22.3       60  2:SPRING   \n",
       "7     3.37  37.2   0.7      0   1: Yes   0: No    37.9     53.6  2:SPRING   \n",
       "8     2.55  45.7     0      0   1: Yes  1: Yes    45.7     50.2  2:SPRING   \n",
       "9     1.93  26.7     0      0   1: Yes  1: Yes    26.7     49.7  2:SPRING   \n",
       "\n",
       "      VDstatus  \n",
       "0  sufficiency  \n",
       "1  sufficiency  \n",
       "2  sufficiency  \n",
       "3  sufficiency  \n",
       "4  sufficiency  \n",
       "5   deficiency  \n",
       "6  sufficiency  \n",
       "7  sufficiency  \n",
       "8  sufficiency  \n",
       "9  sufficiency  \n",
       "\n",
       "[10 rows x 26 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "samples.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "sites = samples.loc[:, 'SITE']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# reference: http://www.latlong.net/\n",
    "Latitude = []\n",
    "Longitude = []\n",
    "sites = samples.loc[:, 'SITE']\n",
    "for i in range(samples.shape[0]):\n",
    "    if sites[i] == 'Birmingham':\n",
    "        latitude = '33.520661'\n",
    "        longitude = '-86.80249'\n",
    "    elif sites[i] == 'San Diego':\n",
    "        latitude = '32.715738'\n",
    "        longitude = '-117.1611'\n",
    "    elif sites[i] == 'Pittsburgh':\n",
    "        latitude = '40.440625'\n",
    "        longitude = '-79.99589'\n",
    "    elif sites[i] == 'Palo Alto':\n",
    "        latitude = '37.441883'\n",
    "        longitude = '-122.143'\n",
    "    elif sites[i] == 'Portland':\n",
    "        latitude = '45.523062'\n",
    "        longitude = '-122.6765'    \n",
    "    elif sites[i] == 'Minneapolis':\n",
    "        latitude = '44.977753'\n",
    "        longitude = '-93.26501'  \n",
    "    Latitude.append(latitude)\n",
    "    Longitude.append(longitude)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1       Birmingham\n",
      "90     Minneapolis\n",
      "200      Palo Alto\n",
      "300     Pittsburgh\n",
      "400       Portland\n",
      "500      San Diego\n",
      "Name: SITE, dtype: object\n",
      "['33.520661' '44.977753' '37.441883' '40.440625' '45.523062' '32.715738']\n",
      "['-86.80249' '-93.26501' '-122.143' '-79.99589' '-122.6765' '-117.1611']\n"
     ]
    }
   ],
   "source": [
    "# simple check\n",
    "print(samples['SITE'][[1,90,200,300,400, 500]])\n",
    "print(np.array(Latitude)[[1,90,200,300,400, 500]])\n",
    "print(np.array(Longitude)[[1,90,200,300,400, 500]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "m = samples.shape[0]\n",
    "required = {'Sample_Name': samples.loc[:,'ID'].values,\n",
    "        'Title': np.repeat('MrOS_VitaminD', m),\n",
    "        'Anonymized_Name': samples.loc[:,'ID'].values,\n",
    "        'Scientific_Name': np.repeat('human gut metagenome', m),\n",
    "        'Taxon_ID': np.repeat('Not applicable', m),\n",
    "        #'Description': np.repeat('Not applicable', m),\n",
    "        'Sample_Type': np.repeat('stool', m),\n",
    "        'Geo_Loc_Name': samples.loc[:, 'SITE'].values,\n",
    "        'Elevation': np.repeat('Not applicable', m),\n",
    "        'Env_Biome': np.repeat('urban biome', m),\n",
    "        'Env_Feature': np.repeat('human-associated habitat', m),\n",
    "        'Env_Material': np.repeat('feces', m),\n",
    "        'Env_Package': np.repeat('human-gut', m),\n",
    "        'Latitude': Latitude,\n",
    "        'Longitude': Longitude,\n",
    "        'Collection_Timestamp': np.repeat('Not applicable', m),\n",
    "        'DNA_Extracted': np.repeat('Not applicable', m),\n",
    "        'Physical_Specimen_Location': np.repeat('Not applicable', m),\n",
    "        'Physical_Specimen_Remaining': np.repeat('Not applicable', m),\n",
    "        'Age': samples.loc[:,'V4AGE1'].values,\n",
    "        'Age_Units': np.repeat('years', m),\n",
    "        'Host_Subject_ID': np.repeat('Not applicable', m),\n",
    "        'Host_Taxid': np.repeat('Not applicable', m),\n",
    "        'Host_Scientific_Name': np.repeat('Homo sapiens', m),\n",
    "        'Host_Common_Name': np.repeat('human', m),\n",
    "        'Life_Stage': np.repeat('adult', m),\n",
    "        'Sex': np.repeat('male', m),\n",
    "        'Height': samples.loc[:, 'HWHGT'].values,\n",
    "        'Height_Units': np.repeat('cm', m),\n",
    "        'Weight': samples.loc[:, 'HWWGT'].values,\n",
    "        'Weight_Units': np.repeat('kg', m),  \n",
    "        'BMI': samples.loc[:, 'HWBMI'].values,\n",
    "        'Body_Habitat': np.repeat('UBERON:feces', m),\n",
    "        'Body_Site': np.repeat('UBERON:feces', m),\n",
    "        'Body_Product': np.repeat('UBERON:feces', m)}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "sample_info = pd.concat([pd.DataFrame(required), samples], axis=1)\n",
    "sample_info = pd.DataFrame(sample_info, columns = ['Sample_Name', 'Title', 'Anonymized_Name',\n",
    "          'Scientific_Name', 'Taxon_ID', \n",
    "          #'Description', \n",
    "          'Sample_Type', 'Geo_Loc_Name', 'Elevation',\n",
    "          'Env_Biome', 'Env_Feature', 'Env_Material', 'Env_Package', 'Latitude', 'Longitude',\n",
    "          'Collection_Timestamp', 'DNA_Extracted', 'Physical_Specimen_Location', 'Physical_Specimen_Remaining',\n",
    "          'Age', 'Age_Units', 'Host_Subject_ID', 'Host_Taxid', 'Host_Scientific_Name', 'Host_Common_Name',\n",
    "          'Life_Stage', 'Sex', 'Height', 'Height_Units', 'Weight', 'Weight_Units', 'BMI', 'Body_Habitat',\n",
    "          'Body_Site', 'Body_Product', 'GIERACE', 'SITE', 'TUDRAMT', 'PASCORE', 'TURSMOKE', 'DTVITD', \n",
    "          'M1ADEPR', 'M1VITMND', 'M1ANTIB', 'M1PROBI', 'OHV1D3', 'OHV24D3', \n",
    "          'OHVD3', 'OHVD2', 'OHV1D2', 'OHV1D2CT', 'OHVD2CT', 'OHVDTOT', 'OHV1DTOT', 'OHSEAS', 'VDstatus'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(599, 55)"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_info.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Name</th>\n",
       "      <th>Title</th>\n",
       "      <th>Anonymized_Name</th>\n",
       "      <th>Scientific_Name</th>\n",
       "      <th>Taxon_ID</th>\n",
       "      <th>Sample_Type</th>\n",
       "      <th>Geo_Loc_Name</th>\n",
       "      <th>Elevation</th>\n",
       "      <th>Env_Biome</th>\n",
       "      <th>Env_Feature</th>\n",
       "      <th>...</th>\n",
       "      <th>OHV24D3</th>\n",
       "      <th>OHVD3</th>\n",
       "      <th>OHVD2</th>\n",
       "      <th>OHV1D2</th>\n",
       "      <th>OHV1D2CT</th>\n",
       "      <th>OHVD2CT</th>\n",
       "      <th>OHVDTOT</th>\n",
       "      <th>OHV1DTOT</th>\n",
       "      <th>OHSEAS</th>\n",
       "      <th>VDstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BI0023</td>\n",
       "      <td>MrOS_VitaminD</td>\n",
       "      <td>BI0023</td>\n",
       "      <td>human gut metagenome</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>stool</td>\n",
       "      <td>Birmingham</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>human-associated habitat</td>\n",
       "      <td>...</td>\n",
       "      <td>1.77</td>\n",
       "      <td>25.8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>25.8</td>\n",
       "      <td>39.3</td>\n",
       "      <td>3:SUMMER</td>\n",
       "      <td>sufficiency</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BI0056</td>\n",
       "      <td>MrOS_VitaminD</td>\n",
       "      <td>BI0056</td>\n",
       "      <td>human gut metagenome</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>stool</td>\n",
       "      <td>Birmingham</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>human-associated habitat</td>\n",
       "      <td>...</td>\n",
       "      <td>3.91</td>\n",
       "      <td>39.2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>39.2</td>\n",
       "      <td>61.9</td>\n",
       "      <td>2:SPRING</td>\n",
       "      <td>sufficiency</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BI0131</td>\n",
       "      <td>MrOS_VitaminD</td>\n",
       "      <td>BI0131</td>\n",
       "      <td>human gut metagenome</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>stool</td>\n",
       "      <td>Birmingham</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>human-associated habitat</td>\n",
       "      <td>...</td>\n",
       "      <td>1.49</td>\n",
       "      <td>23.1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>23.1</td>\n",
       "      <td>52.1</td>\n",
       "      <td>2:SPRING</td>\n",
       "      <td>sufficiency</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BI0153</td>\n",
       "      <td>MrOS_VitaminD</td>\n",
       "      <td>BI0153</td>\n",
       "      <td>human gut metagenome</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>stool</td>\n",
       "      <td>Birmingham</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>human-associated habitat</td>\n",
       "      <td>...</td>\n",
       "      <td>2.14</td>\n",
       "      <td>27.3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>27.3</td>\n",
       "      <td>43.1</td>\n",
       "      <td>2:SPRING</td>\n",
       "      <td>sufficiency</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BI0215</td>\n",
       "      <td>MrOS_VitaminD</td>\n",
       "      <td>BI0215</td>\n",
       "      <td>human gut metagenome</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>stool</td>\n",
       "      <td>Birmingham</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>human-associated habitat</td>\n",
       "      <td>...</td>\n",
       "      <td>3.62</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>33</td>\n",
       "      <td>50.2</td>\n",
       "      <td>4:FALL</td>\n",
       "      <td>sufficiency</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>BI0353</td>\n",
       "      <td>MrOS_VitaminD</td>\n",
       "      <td>BI0353</td>\n",
       "      <td>human gut metagenome</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>stool</td>\n",
       "      <td>Birmingham</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>human-associated habitat</td>\n",
       "      <td>...</td>\n",
       "      <td>1.79</td>\n",
       "      <td>19.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>19.5</td>\n",
       "      <td>45.5</td>\n",
       "      <td>2:SPRING</td>\n",
       "      <td>deficiency</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>BI0371</td>\n",
       "      <td>MrOS_VitaminD</td>\n",
       "      <td>BI0371</td>\n",
       "      <td>human gut metagenome</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>stool</td>\n",
       "      <td>Birmingham</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>human-associated habitat</td>\n",
       "      <td>...</td>\n",
       "      <td>1.82</td>\n",
       "      <td>22.3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>22.3</td>\n",
       "      <td>60</td>\n",
       "      <td>2:SPRING</td>\n",
       "      <td>sufficiency</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>BI0372</td>\n",
       "      <td>MrOS_VitaminD</td>\n",
       "      <td>BI0372</td>\n",
       "      <td>human gut metagenome</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>stool</td>\n",
       "      <td>Birmingham</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>human-associated habitat</td>\n",
       "      <td>...</td>\n",
       "      <td>3.37</td>\n",
       "      <td>37.2</td>\n",
       "      <td>0.7</td>\n",
       "      <td>0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>0: No</td>\n",
       "      <td>37.9</td>\n",
       "      <td>53.6</td>\n",
       "      <td>2:SPRING</td>\n",
       "      <td>sufficiency</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>BI0380</td>\n",
       "      <td>MrOS_VitaminD</td>\n",
       "      <td>BI0380</td>\n",
       "      <td>human gut metagenome</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>stool</td>\n",
       "      <td>Birmingham</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>human-associated habitat</td>\n",
       "      <td>...</td>\n",
       "      <td>2.55</td>\n",
       "      <td>45.7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>45.7</td>\n",
       "      <td>50.2</td>\n",
       "      <td>2:SPRING</td>\n",
       "      <td>sufficiency</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>BI0389</td>\n",
       "      <td>MrOS_VitaminD</td>\n",
       "      <td>BI0389</td>\n",
       "      <td>human gut metagenome</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>stool</td>\n",
       "      <td>Birmingham</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>human-associated habitat</td>\n",
       "      <td>...</td>\n",
       "      <td>1.93</td>\n",
       "      <td>26.7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>26.7</td>\n",
       "      <td>49.7</td>\n",
       "      <td>2:SPRING</td>\n",
       "      <td>sufficiency</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 55 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Sample_Name          Title Anonymized_Name       Scientific_Name  \\\n",
       "0      BI0023  MrOS_VitaminD          BI0023  human gut metagenome   \n",
       "1      BI0056  MrOS_VitaminD          BI0056  human gut metagenome   \n",
       "2      BI0131  MrOS_VitaminD          BI0131  human gut metagenome   \n",
       "3      BI0153  MrOS_VitaminD          BI0153  human gut metagenome   \n",
       "4      BI0215  MrOS_VitaminD          BI0215  human gut metagenome   \n",
       "5      BI0353  MrOS_VitaminD          BI0353  human gut metagenome   \n",
       "6      BI0371  MrOS_VitaminD          BI0371  human gut metagenome   \n",
       "7      BI0372  MrOS_VitaminD          BI0372  human gut metagenome   \n",
       "8      BI0380  MrOS_VitaminD          BI0380  human gut metagenome   \n",
       "9      BI0389  MrOS_VitaminD          BI0389  human gut metagenome   \n",
       "\n",
       "         Taxon_ID Sample_Type Geo_Loc_Name       Elevation    Env_Biome  \\\n",
       "0  Not applicable       stool   Birmingham  Not applicable  urban biome   \n",
       "1  Not applicable       stool   Birmingham  Not applicable  urban biome   \n",
       "2  Not applicable       stool   Birmingham  Not applicable  urban biome   \n",
       "3  Not applicable       stool   Birmingham  Not applicable  urban biome   \n",
       "4  Not applicable       stool   Birmingham  Not applicable  urban biome   \n",
       "5  Not applicable       stool   Birmingham  Not applicable  urban biome   \n",
       "6  Not applicable       stool   Birmingham  Not applicable  urban biome   \n",
       "7  Not applicable       stool   Birmingham  Not applicable  urban biome   \n",
       "8  Not applicable       stool   Birmingham  Not applicable  urban biome   \n",
       "9  Not applicable       stool   Birmingham  Not applicable  urban biome   \n",
       "\n",
       "                Env_Feature  ... OHV24D3 OHVD3 OHVD2 OHV1D2 OHV1D2CT OHVD2CT  \\\n",
       "0  human-associated habitat  ...    1.77  25.8     0      0   1: Yes  1: Yes   \n",
       "1  human-associated habitat  ...    3.91  39.2     0      0   1: Yes  1: Yes   \n",
       "2  human-associated habitat  ...    1.49  23.1     0      0   1: Yes  1: Yes   \n",
       "3  human-associated habitat  ...    2.14  27.3     0      0   1: Yes  1: Yes   \n",
       "4  human-associated habitat  ...    3.62    33     0      0   1: Yes  1: Yes   \n",
       "5  human-associated habitat  ...    1.79  19.5     0      0   1: Yes  1: Yes   \n",
       "6  human-associated habitat  ...    1.82  22.3     0      0   1: Yes  1: Yes   \n",
       "7  human-associated habitat  ...    3.37  37.2   0.7      0   1: Yes   0: No   \n",
       "8  human-associated habitat  ...    2.55  45.7     0      0   1: Yes  1: Yes   \n",
       "9  human-associated habitat  ...    1.93  26.7     0      0   1: Yes  1: Yes   \n",
       "\n",
       "  OHVDTOT OHV1DTOT    OHSEAS     VDstatus  \n",
       "0    25.8     39.3  3:SUMMER  sufficiency  \n",
       "1    39.2     61.9  2:SPRING  sufficiency  \n",
       "2    23.1     52.1  2:SPRING  sufficiency  \n",
       "3    27.3     43.1  2:SPRING  sufficiency  \n",
       "4      33     50.2    4:FALL  sufficiency  \n",
       "5    19.5     45.5  2:SPRING   deficiency  \n",
       "6    22.3       60  2:SPRING  sufficiency  \n",
       "7    37.9     53.6  2:SPRING  sufficiency  \n",
       "8    45.7     50.2  2:SPRING  sufficiency  \n",
       "9    26.7     49.7  2:SPRING  sufficiency  \n",
       "\n",
       "[10 rows x 55 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_info.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Match sample_name in two documents"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "id_prep=list(prep_info.loc[:,'Sample_Name'].values)\n",
    "id_sample=list(sample_info.loc[:,'Sample_Name'].values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "# samples with metadata yet no sequencing data\n",
    "for i in id_sample:\n",
    "    if i not in id_prep:\n",
    "        print(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "# samples with sequencing data yet no metadata\n",
    "no_sample_info = []\n",
    "for j in id_prep:\n",
    "    if j not in id_sample:\n",
    "        no_sample_info.append(j)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n",
      "['PO7100']\n"
     ]
    }
   ],
   "source": [
    "print(len(no_sample_info))\n",
    "print(no_sample_info)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## Drop un-needed samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "# exclude PO7100, as it has no microbiome data (see Lily Liu's email on 08/07/2017)\n",
    "prep_info = prep_info.loc[~prep_info['Sample_Name'].isin(no_sample_info)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(599, 11)"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prep_info.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## preparation and sample information output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "sample_info.to_csv('../data/sample_MrOS.txt', sep= '\\t', na_rep='Missing:not collected', index=False)\n",
    "prep_info.to_csv('../data/prep_MrOS.txt', sep= '\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## create Qiime mapping file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Name</th>\n",
       "      <th>Barcode</th>\n",
       "      <th>LinkerPrimerSequence</th>\n",
       "      <th>Description</th>\n",
       "      <th>Experiment_Design_Description</th>\n",
       "      <th>Library_Construction_Protocol</th>\n",
       "      <th>Linker</th>\n",
       "      <th>Platform</th>\n",
       "      <th>Center_Name</th>\n",
       "      <th>Center_Project</th>\n",
       "      <th>Instrument_Model</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BI0023</td>\n",
       "      <td>TCTGGTGACATT</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0023.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BI0056</td>\n",
       "      <td>CAAGCATGCCTA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0056.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BI0131</td>\n",
       "      <td>CTATTTGCGACA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0131.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BI0153</td>\n",
       "      <td>ATCGGCGTTACA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0153.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BI0215</td>\n",
       "      <td>CCTCTCGTGATC</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0215.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Sample_Name       Barcode  LinkerPrimerSequence       Description  \\\n",
       "0      BI0023  TCTGGTGACATT  GGACTACHVGGGTWTCTAAT  Orwoll.BI0023.BI   \n",
       "1      BI0056  CAAGCATGCCTA  GGACTACHVGGGTWTCTAAT  Orwoll.BI0056.BI   \n",
       "2      BI0131  CTATTTGCGACA  GGACTACHVGGGTWTCTAAT  Orwoll.BI0131.BI   \n",
       "3      BI0153  ATCGGCGTTACA  GGACTACHVGGGTWTCTAAT  Orwoll.BI0153.BI   \n",
       "4      BI0215  CCTCTCGTGATC  GGACTACHVGGGTWTCTAAT  Orwoll.BI0215.BI   \n",
       "\n",
       "                       Experiment_Design_Description  \\\n",
       "0  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "1  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "2  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "3  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "4  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "\n",
       "  Library_Construction_Protocol Linker  Platform Center_Name Center_Project  \\\n",
       "0                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "1                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "2                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "3                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "4                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "\n",
       "  Instrument_Model  \n",
       "0   Illumina MiSeq  \n",
       "1   Illumina MiSeq  \n",
       "2   Illumina MiSeq  \n",
       "3   Illumina MiSeq  \n",
       "4   Illumina MiSeq  "
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prep_info.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "mapping = pd.merge(prep_info, sample_info, on='Sample_Name')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(599, 11)\n",
      "(599, 55)\n",
      "(599, 65)\n"
     ]
    }
   ],
   "source": [
    "print(prep_info.shape)\n",
    "print(sample_info.shape)\n",
    "print(mapping.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "mapping.rename(columns={'Sample_Name': '#SampleID'}, inplace=True)\n",
    "mapping.rename(columns={'Barcode': 'BarcodeSequence'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(599, 65)"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mapping.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#SampleID</th>\n",
       "      <th>BarcodeSequence</th>\n",
       "      <th>LinkerPrimerSequence</th>\n",
       "      <th>Description</th>\n",
       "      <th>Experiment_Design_Description</th>\n",
       "      <th>Library_Construction_Protocol</th>\n",
       "      <th>Linker</th>\n",
       "      <th>Platform</th>\n",
       "      <th>Center_Name</th>\n",
       "      <th>Center_Project</th>\n",
       "      <th>...</th>\n",
       "      <th>OHV24D3</th>\n",
       "      <th>OHVD3</th>\n",
       "      <th>OHVD2</th>\n",
       "      <th>OHV1D2</th>\n",
       "      <th>OHV1D2CT</th>\n",
       "      <th>OHVD2CT</th>\n",
       "      <th>OHVDTOT</th>\n",
       "      <th>OHV1DTOT</th>\n",
       "      <th>OHSEAS</th>\n",
       "      <th>VDstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BI0023</td>\n",
       "      <td>TCTGGTGACATT</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0023.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>...</td>\n",
       "      <td>1.77</td>\n",
       "      <td>25.8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>25.8</td>\n",
       "      <td>39.3</td>\n",
       "      <td>3:SUMMER</td>\n",
       "      <td>sufficiency</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BI0056</td>\n",
       "      <td>CAAGCATGCCTA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0056.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>...</td>\n",
       "      <td>3.91</td>\n",
       "      <td>39.2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>39.2</td>\n",
       "      <td>61.9</td>\n",
       "      <td>2:SPRING</td>\n",
       "      <td>sufficiency</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BI0131</td>\n",
       "      <td>CTATTTGCGACA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0131.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>...</td>\n",
       "      <td>1.49</td>\n",
       "      <td>23.1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>23.1</td>\n",
       "      <td>52.1</td>\n",
       "      <td>2:SPRING</td>\n",
       "      <td>sufficiency</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BI0153</td>\n",
       "      <td>ATCGGCGTTACA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0153.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>...</td>\n",
       "      <td>2.14</td>\n",
       "      <td>27.3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>27.3</td>\n",
       "      <td>43.1</td>\n",
       "      <td>2:SPRING</td>\n",
       "      <td>sufficiency</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BI0215</td>\n",
       "      <td>CCTCTCGTGATC</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0215.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>...</td>\n",
       "      <td>3.62</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>33</td>\n",
       "      <td>50.2</td>\n",
       "      <td>4:FALL</td>\n",
       "      <td>sufficiency</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 65 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  #SampleID BarcodeSequence  LinkerPrimerSequence       Description  \\\n",
       "0    BI0023    TCTGGTGACATT  GGACTACHVGGGTWTCTAAT  Orwoll.BI0023.BI   \n",
       "1    BI0056    CAAGCATGCCTA  GGACTACHVGGGTWTCTAAT  Orwoll.BI0056.BI   \n",
       "2    BI0131    CTATTTGCGACA  GGACTACHVGGGTWTCTAAT  Orwoll.BI0131.BI   \n",
       "3    BI0153    ATCGGCGTTACA  GGACTACHVGGGTWTCTAAT  Orwoll.BI0153.BI   \n",
       "4    BI0215    CCTCTCGTGATC  GGACTACHVGGGTWTCTAAT  Orwoll.BI0215.BI   \n",
       "\n",
       "                       Experiment_Design_Description  \\\n",
       "0  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "1  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "2  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "3  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "4  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "\n",
       "  Library_Construction_Protocol Linker  Platform Center_Name Center_Project  \\\n",
       "0                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "1                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "2                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "3                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "4                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "\n",
       "   ... OHV24D3 OHVD3 OHVD2 OHV1D2 OHV1D2CT OHVD2CT OHVDTOT OHV1DTOT    OHSEAS  \\\n",
       "0  ...    1.77  25.8     0      0   1: Yes  1: Yes    25.8     39.3  3:SUMMER   \n",
       "1  ...    3.91  39.2     0      0   1: Yes  1: Yes    39.2     61.9  2:SPRING   \n",
       "2  ...    1.49  23.1     0      0   1: Yes  1: Yes    23.1     52.1  2:SPRING   \n",
       "3  ...    2.14  27.3     0      0   1: Yes  1: Yes    27.3     43.1  2:SPRING   \n",
       "4  ...    3.62    33     0      0   1: Yes  1: Yes      33     50.2    4:FALL   \n",
       "\n",
       "      VDstatus  \n",
       "0  sufficiency  \n",
       "1  sufficiency  \n",
       "2  sufficiency  \n",
       "3  sufficiency  \n",
       "4  sufficiency  \n",
       "\n",
       "[5 rows x 65 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mapping.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "# move 'Description' to the end\n",
    "cols = list(mapping)\n",
    "cols.insert(mapping.shape[1], cols.pop(cols.index('Description')))\n",
    "mapping = mapping.loc[:, cols]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#SampleID</th>\n",
       "      <th>BarcodeSequence</th>\n",
       "      <th>LinkerPrimerSequence</th>\n",
       "      <th>Experiment_Design_Description</th>\n",
       "      <th>Library_Construction_Protocol</th>\n",
       "      <th>Linker</th>\n",
       "      <th>Platform</th>\n",
       "      <th>Center_Name</th>\n",
       "      <th>Center_Project</th>\n",
       "      <th>Instrument_Model</th>\n",
       "      <th>...</th>\n",
       "      <th>OHVD3</th>\n",
       "      <th>OHVD2</th>\n",
       "      <th>OHV1D2</th>\n",
       "      <th>OHV1D2CT</th>\n",
       "      <th>OHVD2CT</th>\n",
       "      <th>OHVDTOT</th>\n",
       "      <th>OHV1DTOT</th>\n",
       "      <th>OHSEAS</th>\n",
       "      <th>VDstatus</th>\n",
       "      <th>Description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BI0023</td>\n",
       "      <td>TCTGGTGACATT</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>...</td>\n",
       "      <td>25.8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>25.8</td>\n",
       "      <td>39.3</td>\n",
       "      <td>3:SUMMER</td>\n",
       "      <td>sufficiency</td>\n",
       "      <td>Orwoll.BI0023.BI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BI0056</td>\n",
       "      <td>CAAGCATGCCTA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>...</td>\n",
       "      <td>39.2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>39.2</td>\n",
       "      <td>61.9</td>\n",
       "      <td>2:SPRING</td>\n",
       "      <td>sufficiency</td>\n",
       "      <td>Orwoll.BI0056.BI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BI0131</td>\n",
       "      <td>CTATTTGCGACA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>...</td>\n",
       "      <td>23.1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>23.1</td>\n",
       "      <td>52.1</td>\n",
       "      <td>2:SPRING</td>\n",
       "      <td>sufficiency</td>\n",
       "      <td>Orwoll.BI0131.BI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BI0153</td>\n",
       "      <td>ATCGGCGTTACA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>...</td>\n",
       "      <td>27.3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>27.3</td>\n",
       "      <td>43.1</td>\n",
       "      <td>2:SPRING</td>\n",
       "      <td>sufficiency</td>\n",
       "      <td>Orwoll.BI0153.BI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BI0215</td>\n",
       "      <td>CCTCTCGTGATC</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>...</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>33</td>\n",
       "      <td>50.2</td>\n",
       "      <td>4:FALL</td>\n",
       "      <td>sufficiency</td>\n",
       "      <td>Orwoll.BI0215.BI</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 65 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  #SampleID BarcodeSequence  LinkerPrimerSequence  \\\n",
       "0    BI0023    TCTGGTGACATT  GGACTACHVGGGTWTCTAAT   \n",
       "1    BI0056    CAAGCATGCCTA  GGACTACHVGGGTWTCTAAT   \n",
       "2    BI0131    CTATTTGCGACA  GGACTACHVGGGTWTCTAAT   \n",
       "3    BI0153    ATCGGCGTTACA  GGACTACHVGGGTWTCTAAT   \n",
       "4    BI0215    CCTCTCGTGATC  GGACTACHVGGGTWTCTAAT   \n",
       "\n",
       "                       Experiment_Design_Description  \\\n",
       "0  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "1  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "2  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "3  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "4  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "\n",
       "  Library_Construction_Protocol Linker  Platform Center_Name Center_Project  \\\n",
       "0                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "1                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "2                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "3                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "4                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "\n",
       "  Instrument_Model  ... OHVD3 OHVD2 OHV1D2 OHV1D2CT OHVD2CT OHVDTOT OHV1DTOT  \\\n",
       "0   Illumina MiSeq  ...  25.8     0      0   1: Yes  1: Yes    25.8     39.3   \n",
       "1   Illumina MiSeq  ...  39.2     0      0   1: Yes  1: Yes    39.2     61.9   \n",
       "2   Illumina MiSeq  ...  23.1     0      0   1: Yes  1: Yes    23.1     52.1   \n",
       "3   Illumina MiSeq  ...  27.3     0      0   1: Yes  1: Yes    27.3     43.1   \n",
       "4   Illumina MiSeq  ...    33     0      0   1: Yes  1: Yes      33     50.2   \n",
       "\n",
       "     OHSEAS     VDstatus       Description  \n",
       "0  3:SUMMER  sufficiency  Orwoll.BI0023.BI  \n",
       "1  2:SPRING  sufficiency  Orwoll.BI0056.BI  \n",
       "2  2:SPRING  sufficiency  Orwoll.BI0131.BI  \n",
       "3  2:SPRING  sufficiency  Orwoll.BI0153.BI  \n",
       "4    4:FALL  sufficiency  Orwoll.BI0215.BI  \n",
       "\n",
       "[5 rows x 65 columns]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mapping.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "mapping.to_csv('../data/mapping_MrOS.txt', sep= '\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
